runtime/util/thrhelp.c

```diff
@@ -37,3 +38,24 @@ getVMThreadFromOMRThread(J9JavaVM *vm, omrthread_t omrthread)
 	}
 	return vmThread;
 }
+
+void
+initializeCurrentOSStackFree(J9VMThread *vmThread, omrthread_t osThread, UDATA osStackSize)
+{
+	UDATA actualStackSize = 0;
+	UDATA stackStart = 0;
+	UDATA stackEnd = 0;
+
+	if (J9THREAD_SUCCESS == omrthread_get_stack_range(osThread, (void**)&stackStart, (void **)&stackEnd)) {
+		actualStackSize = stackEnd - stackStart;
+		vmThread->currentOSStackFree = ((UDATA)&stackStart - stackStart);
+	} else {
+		UDATA osStackFree = omrthread_current_stack_free();
+		if (0 != osStackFree) {
+			vmThread->currentOSStackFree = osStackFree - (osStackFree / J9VMTHREAD_RESERVED_C_STACK_FRACTION);
+		} else {
+			vmThread->currentOSStackFree = osStackSize - (osStackSize / J9VMTHREAD_RESERVED_C_STACK_FRACTION);
+		}
+	}
+	Trc_Util_thrhelp_initializeCurrentOSStackFree(vmThread, osThread, osStackSize, actualStackSize, vmThread->currentOSStackFree, &actualStackSize);
```

Comment:
I think vmThread needs to be passed in twice. Once for the trace framework, and again so it can be printed. Alternatively you can remove `vmthread=%p` from the tracepoint, since the trace framework includes the current thread.

Reply:
I think this is still incorrect - the tracepoint has been declared noEnv so the vmThread should not be passed at all (or the tracepoint should not be declared noEnv, which would be faster if you always have the vmThread around).